capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java

```diff
@@ -128,4 +135,68 @@ public static void createUser(int userId, String userName, String email) throws
     }
     return itemTypes;
   }
+
+  /*
+   * Provided a list of ItemTypes, this method will return a list of Stores
+   * with every permutation of possible items from any given store that 
+   * satisfies this list of ItemTypes.
+   *
+   * @param itemTypes list of Strings that describes the desired Item Types
+   * @return a list of all potential stores and combinations that satisfy the Item Types
+   *
+   */
+  public static List<Store> getStoresWithItems(List<String> itemTypes) {
+    List<Store> stores = new ArrayList<Store>();
+    DatabaseClient dbClient = initClient();
+    String query = "";
+    boolean first = true;
+    for (String itemType : itemTypes) {
+      List<Store> newStores = new ArrayList<Store>();
+      query = "SELECT ItemId, ItemNameAndBrand FROM Items WHERE ItemType=@itemType";
```

Comment:
can just do 1 query? For example `SELECT a.ItemId, a.ItemNameAndBrand, b.StoreId, b.Price, From Items a JOIN Inventory b ON a.ItemId = b.ItemId WHERE b.ItemAvailabilty = "AVAILABLE" AND a.ItemType IN @itemType` (and you can pass in a list for itemType). And you can join with the Stores table too.

Ideally you should only execute 1 query to get all your data (because database access is really slow), and joining tables is quick